proposals/first-class-span-types.md

```diff
@@ -62,37 +62,46 @@ as if the `T` was declared as `out T` in some scenarios. We do not, however, plu
 variance-convertible in [§18.2.3.3](https://github.com/dotnet/csharpstandard/blob/draft-v8/standard/interfaces.md#18233-variance-conversion). If in the future, we change the runtime
 to more deeply understand the variance here, we can take the minor breaking change to fully recognize it in the language.
 
-Practically, this will also mean that in pattern matching for generic scenarios, we'd have behavior as follows:
+There is also an open question below about participation in delegate signature matching.
 
-```cs
-using System;
+#### Patterns
 
-M<object[]>(["0"]); // Does not print
-M<ReadOnlySpan<string>>(["1"]); // Does not print
-M<Span<object>>(["2"]); // Does not print
-M<ReadOnlySpan<object>>(["3"]); // Prints
+Note that `ref struct`s are not supported in patterns, except for trivial identity patterns:
 
-void M<T>(T t) where T : allows ref struct
+```cs
+class C<T> where T : allows ref struct
 {
-    if (t is ReadOnlySpan<object> r) Console.WriteLine(r[0]);
+    void M1(T t) { if (t is T x) { } } // ok
+    void M2(R r) { if (r is R x) { } } // ok
+    void M3(T t) { if (t is R x) { } } // error
+    void M4(R r) { if (r is T x) { } } // error
 }
+ref struct R { }
```

Comment:
@AaronRobinsonMSFT wanted to double check with you that this is not intended to work at the IL level with your changes.

Reply:
Correct.